File: nexapp/nexapp/report/employee_details_report/employee_details_report.py
# Copyright (c) 2026, Nexapp and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)
	return columns, data

def get_columns():
	return [
		{
			"fieldname": "employee_name",
			"label": _("Full Name"),
			"fieldtype": "Data",
			"width": 180
		},
		{
			"fieldname": "gender",
			"label": _("Gender"),
			"fieldtype": "Link",
			"options": "Gender",
			"width": 100
		},
		{
			"fieldname": "date_of_joining",
			"label": _("Date of Joining"),
			"fieldtype": "Date",
			"width": 120
		},
		{
			"fieldname": "designation",
			"label": _("Designation"),
			"fieldtype": "Link",
			"options": "Designation",
			"width": 160
		},
		{
			"fieldname": "branch",
			"label": _("Branch"),
			"fieldtype": "Link",
			"options": "Branch",
			"width": 140
		},
		{
			"fieldname": "department",
			"label": _("Department"),
			"fieldtype": "Link",
			"options": "Department",
			"width": 160
		},
		{
			"fieldname": "reports_to",
			"label": _("Reports to"),
			"fieldtype": "Link",
			"options": "Employee",
			"width": 160
		},
		{
			"fieldname": "employment_type",
			"label": _("Employment Type"),
			"fieldtype": "Link",
			"options": "Employment Type",
			"width": 140
		},
		{
			"fieldname": "grade",
			"label": _("Grade"),
			"fieldtype": "Link",
			"options": "Employee Grade",
			"width": 120
		},
		{
			"fieldname": "custom_sub_grade",
			"label": _("Sub Grade"),
			"fieldtype": "Data",
			"width": 120
		},
		{
			"fieldname": "custom_level",
			"label": _("Level"),
			"fieldtype": "Link",
			"options": "Employee Level",
			"width": 120
		},
		{
			"fieldname": "status",
			"label": _("Status"),
			"fieldtype": "Select",
			"width": 100
		}
	]

def get_data(filters):
	query_filters = {}
	if isinstance(filters, dict):
		for key, val in filters.items():
			if val:
				query_filters[key] = val

	data = frappe.db.get_all("Employee",
		fields=[
			"employee_name", "gender", "date_of_joining", "designation",
			"branch", "department", "reports_to", "employment_type",
			"grade", "custom_sub_grade", "custom_level", "status"
		],
		filters=query_filters,
		order_by="employee_name"
	)
	return data

File: nexapp/nexapp/report/employee_details_report/employee_details_report.js
// Copyright (c) 2026, Nexapp and contributors
// For license information, please see license.txt

frappe.query_reports["Employee Details Report"] = {
    "filters": [
        {
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "default": frappe.defaults.get_user_default("Company")
        },
        {
            "fieldname": "status",
            "label": __("Status"),
            "fieldtype": "Select",
            "options": "\nActive\nInactive\nSuspended\nLeft",
            "default": "Active"
        },
        {
            "fieldname": "department",
            "label": __("Department"),
            "fieldtype": "Link",
            "options": "Department"
        },
        {
            "fieldname": "branch",
            "label": __("Branch"),
            "fieldtype": "Link",
            "options": "Branch"
        },
        {
            "fieldname": "designation",
            "label": __("Designation"),
            "fieldtype": "Link",
            "options": "Designation"
        }
    ]
};
